osquery/events/darwin/endpointsecurity.cpp

```diff
@@ -116,7 +116,11 @@ void EndpointSecurityPublisher::handleMessage(const es_message_t* message) {
       for (auto i = 0; i < ec->argc; i++) {
         auto arg = es_exec_arg(&message->event.exec, i);
         auto s = getStringFromToken(&arg);
-        args << s << ' ';
+        if (s.find(" ") != std::string::npos) {
+          args << std::quoted(s) << ' ';
+        } else {
```

Comment:
Ahh good point, we need to quote if there's a `"` too, to know that we quoted 😅 

As long as the string can be parsed unambiguously, it doesn't seem worth the complexity of trying to make this more runnable by a real shell (like trying to escaping `$` within the `"`), so I don't think we need to quote for other characters? Though if we switched to quoting using `'`, it'd be a lot closer (issue then is `'\'test'` isn't valid because single quotes can't be escaped inside single quotes in bash/zsh).

Reply:
I don't think any additional quoting should exist a part from using `"` to hint the fact that space separated words were a single argument.
The representation there is what the program (or the exec syscall) receives, which is not what you would see in the shell.

Reply:
Ok yeah I agree! I think what the PR is doing now (use `std::quoted` whenever there is either a `"` or ` ` in the string) should accomplish this in a reasonable way, but let me know if I'm misunderstanding your point.